packages/markup-frontstage/src/services/SavedView.ts

```diff
@@ -147,22 +220,97 @@ export const createSavedViewData = (vp: Viewport): SavedViewData => {
 
   const data: SavedViewData = {
     alwaysDrawn,
-    neverDrawn,
     categories,
+    changeSetId,
+    displayStyleProps: jsonStringify(viewState.displayStyle?.toJSON()),
+    iModelId,
+    is2d: false,
     models,
-    version,
+    neverDrawn,
+    perModelCategoryVisibility,
     projectId,
-    iModelId,
-    changeSetId,
     sourceId,
-    userView: true,
     state,
-    displayStyleProps: JSON.stringify(viewState.displayStyle.toJSON()),
-    perModelCategoryVisibility,
+    userView: true,
+    version,
   };
 
   // Parse all data from emphasize elements (e.g. colorization, isolates, hidden, emphasize)
   parseFromEmphasizeElements(vp, data);
 
   return data;
 };
+
+/**
+ * Determines whether or not given saved view is spatial.
+ * @param view name of the Saved View.
+ */
+export const isSpatialSavedView = (view: SavedViewData) => {
+  // view.is2d can be undefined for spatialSavedView that are added before 2D saved view feature is added.
+  return !!view.is2d && "models" in view;
```

Comment:
did you mean to add the double negation here?

Reply:
yes sir!